src/azure-cli/azure/cli/command_modules/eventgrid/custom.py

```diff
@@ -1621,6 +1635,24 @@ def _get_sku(sku_name):
     return result
 
 
+def _get_identity_type_with_checks(
+        identity_type_name=IDENTITY_NONE,
+        user_identity_properties=None,
+        mi_system_assigned=None):
+    if identity_type_name is not None and user_identity_properties is None and mi_system_assigned is None:
+        result = _get_identity_type(identity_type_name)
+    elif identity_type_name is None and user_identity_properties is None and mi_system_assigned is not None:
+        result = IDENTITY_SYSTEM_ASSIGNED
+    elif identity_type_name is None and user_identity_properties is not None and mi_system_assigned is None:
+        result = IDENTITY_USER_ASSIGNED
+    elif identity_type_name is None and user_identity_properties is not None and mi_system_assigned is not None:
+        result = IDENTITY_MIXED_MODE
+    elif identity_type_name is not None and (user_identity_properties is not None or mi_system_assigned is not None):
+        raise CLIError('usage error: cannot use --identity together with --mi-system-assigned or --mi-user-assigned')
```

Comment:
Please use the specific error type `MutuallyExclusiveArgumentError` instead of `CLIError`